packages/braintree-integration/src/braintree-fastlane/braintree-fastlane-utils.ts

```diff
@@ -0,0 +1,541 @@
+                billingAddresses,
+            );
+
+            this.browserStorage.setItem('sessionId', cart.id);
+            await this.paymentIntegrationService.updatePaymentProviderCustomer({
+                authenticationState,
+                addresses,
+                instruments,
+            });
+
+            if (billingAddresses.length > 0) {
+                await this.paymentIntegrationService.updateBillingAddress(billingAddresses[0]);
+            }
+
+            if (shippingAddresses.length > 0 && cart.lineItems.physicalItems.length > 0) {
+                await this.paymentIntegrationService.updateShippingAddress(shippingAddresses[0]);
+            }
+        } catch (error) {
+            // TODO: we should figure out what to do here
+            // TODO: because we should not to stop the flow if the error occurs on paypal side
+        }
+    }
+
+    /**
+     *
+     * PayPal to BC data mappers
+     *
+     * */
+    private mapPayPalToBcAddress(
+        addresses?: BraintreeConnectAddress[],
+        phones?: BraintreeConnectPhone[],
+    ): CustomerAddress[] | undefined {
+        if (!addresses) {
+            return;
+        }
+
+        const countries = this.paymentIntegrationService.getState().getCountries() || [];
+        const phoneNumber =
+            phones && phones[0] ? phones[0].country_code + phones[0].national_number : '';
+
+        const getCountryNameByCountryCode = (countryCode: string) => {
+            const matchedCountry = countries.find((country) => country.code === countryCode);
+
+            return matchedCountry?.name || '';
+        };
+
+        return addresses.map((address) => ({
+            id: Number(address.id),
+            type: 'paypal-address',
+            firstName: address.firstName || '',
+            lastName: address.lastName || '',
+            company: address.company || '',
+            address1: address.streetAddress,
+            address2: address.extendedAddress || '',
+            city: address.locality,
+            stateOrProvince: address.region,
+            stateOrProvinceCode: address.region,
+            country: getCountryNameByCountryCode(address.countryCodeAlpha2),
+            countryCode: address.countryCodeAlpha2,
+            postalCode: address.postalCode,
+            phone: phoneNumber,
+            customFields: [],
+        }));
+    }
+
+    private mapPayPalFastlaneToBcAddress(
+        shippingAddress?: BraintreeFastlaneAddress,
+    ): CustomerAddress[] | undefined {
+        if (!shippingAddress) {
+            return;
+        }
+
+        const countries = this.paymentIntegrationService.getState().getCountries() || [];
+
+        const getCountryNameByCountryCode = (countryCode: string) => {
+            const matchedCountry = countries.find((country) => country.code === countryCode);
+
+            return matchedCountry?.name || '';
+        };
+
+        return [
+            {
+                id: Number(shippingAddress.id),
```

Comment:
Same as above

Reply:
You are right there is no `id` field in shippingAddress